compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -465,6 +468,54 @@ fn expand_preparsed_asm(ecx: &mut ExtCtxt<'_>, sp: Span, args: AsmArgs) -> P<ast
         for piece in unverified_pieces {
             match piece {
                 parse::Piece::String(s) => {
+                    if let Some(idx) = s.find(".intel_syntax") {
+                        let mut end = idx + ".intel_syntax".len();
+                        if let Some(prefix_idx) = s.split_at(end).1.find("noprefix") {
+                            // Should be a space and it should be immediately after
+                            if prefix_idx == 1 {
+                                end += " noprefix".len();
+                            }
+                        }
+
+                        let syntax_span =
+                            template_span.from_inner(InnerSpan::new(idx + 1, end + 1));
+                        let mut err = ecx.struct_span_err(syntax_span, "intel sytnax is the default syntax, and trying to use this directive may cause issues");
+                        err.span_suggestion(
+                            syntax_span,
+                            "Remove this assembler directive",
+                            s.replace(&s[idx..end], "").to_string(),
+                            Applicability::MachineApplicable,
+                        );
+                        err.emit();
+                    }
+
+                    if let Some(idx) = s.find(".att_syntax") {
```

Comment:
It would be nice if this and the branch above could be deduplicated into a function.